cypress/support/hub-commands.ts

```diff
@@ -1,18 +1,112 @@
 /* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access */
+import { SetRequired } from 'type-fest';
 import { randomString } from '../../framework/utils/random-string';
 import { Role } from '../../frontend/hub/access/roles/Role';
 import { RemoteRegistry } from '../../frontend/hub/administration/remote-registries/RemoteRegistry';
+import { HubRemote } from '../../frontend/hub/administration/remotes/Remotes';
+import { Repository } from '../../frontend/hub/administration/repositories/Repository';
 import { Task } from '../../frontend/hub/administration/tasks/Task';
 import { CollectionVersionSearch } from '../../frontend/hub/collections/Collection';
 import { parsePulpIDFromURL } from '../../frontend/hub/common/api/hub-api-utils';
 import { HubItemsResponse } from '../../frontend/hub/common/useHubView';
 import { ExecutionEnvironment as HubExecutionEnvironment } from '../../frontend/hub/execution-environments/ExecutionEnvironment';
+import { HubNamespace } from '../../frontend/hub/namespaces/HubNamespace';
 import { galaxykitPassword, galaxykitUsername } from './e2e';
 import { hubAPI, pulpAPI } from './formatApiPathForHub';
-import { escapeForShellCommand } from './utils';
+import { escapeForShellCommand, randomE2Ename } from './utils';
 
 const apiPrefix = Cypress.env('HUB_API_PREFIX') as string;
 
+export interface HubRequestOptions {
+  method: 'DELETE' | 'GET' | 'PATCH' | 'POST' | 'PUT';
+  url: string;
+  body?: object;
+  qs?: object;
+  failOnStatusCode?: boolean;
+}
+Cypress.Commands.add('hubRequest', (options: HubRequestOptions) => {
+  cy.getCookie('csrftoken').then((cookie) =>
+    cy
+      .request({
+        ...options,
+        headers: { 'X-CSRFToken': cookie?.value, Referer: Cypress.config().baseUrl },
+      })
+      .then((response) => {
+        switch (response.status) {
+          case 202: // Accepted
```

Comment:
@jamestalton a couple of questions here:

Do we want to return the task in a few cases?  Perhaps, we could add a option to return the response without waiting for the task to be completed. Mainly thinking about the tasks tests.

We could return the response not matter what the status code?